domain_model/Ballot.ts

```diff
@@ -12,6 +13,7 @@ export interface Ballot {
     ip_address?: string; // ip address if once_per_ip is enabled
     votes: Vote[];         // One per poll
     history?: BallotAction[];
+    precinct?: string; // Precint of voter
```

Comment:
double checking there's no such thing as a voter being in two separate precincts?  (maybe a state-level precinct and a county-level precinct or some craziness?)

Reply:
The way I've been looking at it is the precinct is the lowest level group that voters can be divided into, meaning everyone in that precinct gets the same ballot. You can have situations where the state senate, state assembly, counties, cities, etc., don't share the same district borders. So your neighbor could be in the same state assembly district but different senate district for example. 

When I was looking through the Oregon precinct database it looks like they had to further divide their precincts into sub-precincts which was confusing, but for simpler elections this should cover most use cases.